cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -259,7 +260,7 @@ func (ht *hcsTask) CreateExec(ctx context.Context, req *task.ExecProcessRequest,
 	if err != nil {
 		return err
 	}
-	he := newHcsExec(ctx, ht.events, ht.id, ht.c, req.ExecID, ht.init.Status().Bundle, ht.isWCOW, spec, io)
+	he := newHcsExec(ctx, ht.events, ht.id, ht.host, ht.c, req.ExecID, ht.init.Status().Bundle, ht.isWCOW, spec, io)
```

Comment:
Beginning to think the entire `ht` should just be passed. But not a problem for this PR

Reply:
Yea. I cant really decide on this. Lets chat about the lifetime responsibility. Basically who should track the lifetime, the exec or the task? If we decide on that we can decide on the model but I couldn't quite decide because each has trade offs.